### ISSUE TITLE
"Invalid argument" for json_contains() queries

### ISSUE THREAD

**Issue by dphil** [dolthub/dolt#3895]
Dolt version 0.40.19

To reproduce:
1. Create a table with a JSON column (unsure if relevant, but default it to an empty array `'[]'`).
2. `select * from assets where json_contains(jsonFieldName, '"someStringValue"');` (you don't need any rows in the table, just enter any string to query on)

Results:

In mysql:
```
mysql> select * from test_table where json_contains(json_field, '"hello"');
Empty set (0.00 sec)
```
In dolt:
```
test_db> select * from test_table where json_contains(json_field, '"hello"');
Invalid argument to {{0xc000200580 0x34a7278 [22 5 0 1 2 36 97 50 98 48 49 101 51 50 45 52 53 48 52 45 52 55 51 51 45 97 101 51 54 45 51 101 51 53 99 51 99 101 57 56 102 98] []}}
```

**Comment by fulghum** [comment]
Thanks for the nice repro case; I'm digging in now...

**Comment by fulghum** [comment]
Just a quick update... I was able to repro the issue very quickly thanks to your details and I've been digging through the `json_contains` code and tests. I found some gaps in the implementation and have this case working now. I just need to finishing auditing the code and tests to make sure there aren't any other gaps, then I'll push out a PR for review.

**Comment by fulghum** [comment]
I added a bunch more test cases, did some testing/validation against MySQL, and fixed some additional places where I found Dolt's implementation of `json_contains` was returning incorrect results. I've got a PR out now and will get a review first thing Monday morning and happy to run a release for you after this gets merged in. 

I'm assuming you'll be busy enjoying the weekend, but if for any reason you do need to get your hands on a build with this fix, just ping me and I'll be happy to help you build a pre-release build from my PR branch. 

Thanks again for reporting this one so we could get this fixed! 🙏

**Comment by dphil** [comment]
Thanks @fulghum! I can wait until next week. I'll but out Monday as well, but will check for updates on Tuesday.

**Comment by fulghum** [comment]
I got the fix reviewed and merged in to go-mysql-server. I'll get it into Dolt next and then I'll update this thread one more time later this afternoon when the release is out for you.

**Comment by fulghum** [comment]
Alright, [Dolt version 0.40.20](https://github.com/dolthub/dolt/releases/tag/v0.40.20) is out and contains bug fixes for `json_contains`. 🎉  

Try out that latest version when you get a chance and let us know if you hit any snags.

**Comment by dphil** [comment]
Confirmed this is fixed on our end with version 0.40.20. Thanks!
